core/state_processor_rip7560.go

```diff
@@ -165,22 +166,43 @@ func BuyGasRip7560Transaction(chainConfig *params.ChainConfig, gp *GasPool, head
 }
 
 func ApplyRip7560ValidationPhases(chainConfig *params.ChainConfig, bc ChainContext, author *common.Address, gp *GasPool, statedb *state.StateDB, header *types.Header, tx *types.Transaction, cfg vm.Config) (*ValidationPhaseResult, error) {
-	/*** Nonce Validation Frame ***/
-	nonceValidationMsg := prepareNonceValidationMessage(tx, chainConfig)
 	blockContext := NewEVMBlockContext(header, bc, author, chainConfig, statedb)
-	txContext := NewEVMTxContext(nonceValidationMsg)
-	txContext.Origin = *tx.Rip7560TransactionData().Sender
+	txContext := vm.TxContext{
+		Origin:   *tx.Rip7560TransactionData().Sender,
+		GasPrice: tx.GasFeeCap(),
```

Comment:
This is something we set up in the prepare message step, so I'm curious as to why you explicitly added it here

Reply:
This was because the revised Nonce validation frame can return nil message at prepareNonceValidationMessage. So I revised this to be more clear.